starport_template/lib/widgets/send_money_sheet.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:cosmos_ui_components/components/template/cosmos_wallets_list_view.dart';
+import 'package:flutter/material.dart';
+import 'package:starport_template/entities/amount.dart';
+import 'package:starport_template/entities/balance.dart';
+import 'package:starport_template/entities/denom.dart';
+import 'package:starport_template/starport_app.dart';
+import 'package:transaction_signing_gateway/model/wallet_public_info.dart';
+
+class SendMoneySheet extends StatefulWidget {
+  final Denom denom;
+  final WalletInfo walletInfo;
+
+  const SendMoneySheet({
+    Key? key,
+    required this.denom,
+    required this.walletInfo,
+  }) : super(key: key);
+
+  @override
+  _SendMoneySheetState createState() => _SendMoneySheetState();
+}
+
+class _SendMoneySheetState extends State<SendMoneySheet> {
+  String _toAddress = '';
+  String _amount = '';
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        const Padding(padding: EdgeInsets.only(top: 16)),
+        Text(
+          widget.denom.text,
+          style: Theme.of(context).textTheme.headline6,
+        ),
+        ListTile(
+          title: TextFormField(
+            decoration: const InputDecoration(
+              labelText: 'Enter wallet address',
+              border: OutlineInputBorder(),
+            ),
+            onChanged: (value) {
+              _toAddress = value;
+            },
+          ),
+        ),
+        ListTile(
+          title: TextFormField(
+            decoration: const InputDecoration(
+              labelText: 'Enter amount',
+              border: OutlineInputBorder(),
+            ),
+            onChanged: (value) {
+              _amount = value;
+            },
+          ),
+        ),
+        ElevatedButton(
+          onPressed: () async {
+            final amount = Amount.fromString(_amount);
+            final info = WalletPublicInfo(
+              name: widget.walletInfo.name,
+              publicAddress: widget.walletInfo.address,
+              walletId: widget.walletInfo.address,
```

Comment:
not sure if you can use address as walletId.

Reply:
What do we specifically use `walletId` for?

Reply:
Since we can have multiple wallets specified for the same chain, we use the identifier in order to distinguish wallets from each other. in the template, we generate the walletId when storing the wallet credentials using Uuid package:
https://github.com/tendermint/flutter/blob/88af54dde521e9b58aa02dab4dba90632e5a38af/starport_template/lib/stores/wallets_store.dart#L39. 

So in order to properly send the amount you should retrieve the wallet id from the wallet details

Reply:
✅